packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -1274,25 +1157,21 @@ function transactionProxy<T>(thing: T, transactionId: string): T {
       // we don't want to allow any calls to our `forbidden` methods
       if (forbidden.includes(prop as string)) return undefined
 
-      // secret accessor to get the `transactionId` in a transaction
-      if (prop === TX_ID) return transactionId
+      if (prop === TX_ID) return txId // secret accessor to the txId
 
+      // we override and handle every function call within the proxy
       if (typeof target[prop] === 'function') {
-        // we override & handle every function call within the proxy
         return (...args: unknown[]) => {
-          if (prop === 'then') {
-            // this is our promise, we pass it an extra info argument
-            // this will call "our" `then` which will call `_request`
-            return target[prop](...args, transactionId)
-          }
+          // we hijack `then` calls to pass txId to prisma promises
+          if (prop === 'then') return target[prop](args[0], args[1], txId)
```

Comment:
So if there is only a `.catch()` we don't execute. But it's an internal detail maybe and not user facing?

Reply:
That's a good point, I originally modeled this after the existing transactions https://github.com/prisma/prisma/blob/b0af104d16accd2ed92b0ca1ff73847db1cce3b0/packages/client/src/runtime/core/request/createPrismaPromise.ts/#L37-L42
But since the user can actually call `catch` and `finally`, I will fix and add tests for it.
I guess no one has run into this, as most only use `await`, but I'm going to improve this.

Reply:
I added tests for this.